sonar-plugin-api/src/main/java/org/sonar/api/batch/sensor/SensorDescriptor.java

```diff
@@ -70,18 +72,28 @@
   /**
    * Property this {@link Sensor} depends on. Used by the platform to skip execution of the {@link Sensor} when
    * property is not set.
+   * @deprecated since 6.5 use {@link #onlyWhenConfiguration(Predicate)}
    */
+  @Deprecated
   SensorDescriptor requireProperty(String... propertyKey);
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)